dataclasses_avroschema/model_generator/generator.py

```diff
@@ -285,57 +302,28 @@ def parse_union(self, *, field_types: typing.List, model_name: str) -> str:
             field_types: List of avro types
             model_name: name of the model that contains the `union`
         """
-        first_type = field_types[0]
 
-        if first_type == field_utils.NULL:
-            # It is an optional field, we should include in the imports typing
-            # an use the optional Template
-            self.imports.add("import typing")
-
-            # the union can have native types or custom types
-            # if the type is not in the AVRO_TYPE_TO_PYTHON it means that it is a CustomType
-            if len(field_types) > 2:
-                language_types = self.parse_union(field_types=field_types[1:], model_name=model_name)
-            elif isinstance(field_types[1], dict):
-                # TODO: check what happens with more than 2 complex types
-                language_types = self.render_field(field=field_types[1], model_name=model_name)
+        # XXX: Maybe more useful in general
+        def render_type(typ: str) -> str:
+            if isinstance(typ, dict):
+                return self.render_field(field=typ, model_name=model_name)
             else:
-                language_types = ", ".join(
-                    # TODO: check default, seems doing nothing
-                    [self.get_language_type(type=type, default=type, model_name=model_name) for type in field_types[1:]]
-                )
+                return self.get_language_type(type=typ, model_name=model_name)
 
+        if field_utils.NULL in field_types and len(field_types) == 2:
```

Comment:
This now behaves as follows:

1. `union { null, X }` and `union { X, null }` will get rendered as `Optional[X]`
2. `union { X }` will get rendered as `X`.
3. All other Unions will get rendered as a regular Union.

Reply:
Nice!